lib/rule-tester/rule-tester.js

```diff
@@ -991,81 +992,89 @@ class RuleTester {
                             assert.strictEqual(message.endColumn, error.endColumn, `Error endColumn should be ${error.endColumn}`);
                         }
 
+                        assert.ok(!message.suggestions || hasOwnProperty(error, "suggestions"), `Error at index ${i} has suggestions. Please specify 'suggestions' property on the test error object.`);
                         if (hasOwnProperty(error, "suggestions")) {
 
                             // Support asserting there are no suggestions
-                            if (!error.suggestions || (Array.isArray(error.suggestions) && error.suggestions.length === 0)) {
-                                if (Array.isArray(message.suggestions) && message.suggestions.length > 0) {
-                                    assert.fail(`Error should have no suggestions on error with message: "${message.message}"`);
-                                }
+                            if (!message.suggestions) {
+                                assert.ok(!error.suggestions || (Array.isArray(error.suggestions) && error.suggestions.length === 0), `Error should have suggestions on error with message: "${message.message}"`);
+                            } else if (!error.suggestions || Array.isArray(error.suggestions) && error.suggestions.length === 0) {
+                                assert.ok(message.suggestions.length === 0, `Error should have no suggestions on error with message: "${message.message}"`);
                             } else {
```

Comment:
```suggestion
                            if (!error.suggestions || (Array.isArray(error.suggestions) && error.suggestions.length === 0)) {
                                if (message.suggestions) {
                                    assert.fail(`Error should have no suggestions on error with message: "${message.message}"`);
                                }
                            } else {
                                if (!message.suggestions) {
                                    assert.fail(`Error should have suggestions on error with message: "${message.message}"`);
                                }
```

I think we can simplify this to avoid duplicating the `!error.suggestions || (Array.isArray(error.suggestions) && error.suggestions.length === 0` condition.